kernel-macro/src/lib.rs

```diff
@@ -2,25 +2,43 @@ use proc_macro::TokenStream;
 use quote::quote;
 use syn::{parse_macro_input, ItemFn};
 
+// procedural macros allow to define custom attribute-like syntax
+// and perform code transformations at compile-time
 #[proc_macro_attribute]
+// this is a function signature of procedural macro attribute
+// it takes 2 paramters, representing the attribute arguments
+// and annotated function's code and return a TokenStream
 pub fn main(_: TokenStream, input: TokenStream) -> TokenStream {
-    // Parse the input tokens into a syntax tree representing a function
+    // Take the input token stream and performs the neccessary parsing
+    // and validation to convert it into a Rust syntax tree.
     let input_fn = parse_macro_input!(input as ItemFn);
 
-    // Extract the name of the input function
+    // Extract the name of the input function by accessing its signatures
+    // sig and then retrieving the identifier 'ident'
+    // the clone method is used to create a clone of the function name
     let fn_name = input_fn.sig.ident.clone();
 
+    // quote macro to generate the tranformed code.
```

Comment:
`transformed`, there is a missing "s"

Reply:
fix at this [commit](3d56b448ce27ff2ee25b9e46e3b86aecbbb74896)